reference-implementation/to-upstream-wpts/writable-streams/aborting.js

```diff
@@ -43,5 +41,256 @@ promise_test(t => {
     assert_not_equals(writer.ready, readyPromise, 'the ready promise property should change');
     return promise_rejects(t, new TypeError(), writer.ready, 'the ready promise should reject with a TypeError');
   });
-
 }, 'Aborting a WritableStream should cause the writer\'s fulfilled ready promise to reset to a rejected one');
+
+function promise_fulfills(expectedValue, promise, msg) {
+  promise.then(value => {
+    assert_equals(value, expectedValue, msg);
+  }, reason => {
+    throw new Error(msg + ': Rejected unexpectedly with: ' + reason);
+  });
+}
+
+promise_test(() => {
+  const ws = new WritableStream({});
+
+  const writer = ws.getWriter();
+  writer.releaseLock();
+
+  const abortPromise = writer.abort();
+  return abortPromise.then(() => {
+    throw new Error('abortPromise fulfilled unexpectedly');
+  },
+  r => {
+    assert_equals(r.constructor.toString(), TypeError.toString(), 'abort() should reject with a TypeError');
+  });
+}, 'abort() on a released writer rejects');
+
+promise_test(() => {
+  const ws = new WritableStream({
+    write() {
+      throw new Error('Unexpected write() call');
+    }
+  });
+
+  return delay(0).then(() => {
+    const writer = ws.getWriter();
+
+    writer.abort();
+    writer.write(1);
+    writer.write(2);
+  })
+  .then(() => delay(100));
+}, 'Aborting a WritableStream immediately prevents future writes');
+
+promise_test(() => {
+  let writeCount = 0;
+
+  const ws = new WritableStream({
```

Comment:
Another good example of where recordingWritableStream would help you. You wouldn't need writeCount or any asserts, just check the events in a final `.then` block.